src/commands/whoami/mod.rs

```diff
@@ -1,19 +1,32 @@
 use crate::http;
 use crate::settings::global_user::GlobalUser;
-use crate::terminal::emoji;
+use crate::terminal::{emoji, message};
 
 use cloudflare::endpoints::account::{self, Account};
+use cloudflare::endpoints::user::GetUserDetails;
 use cloudflare::framework::apiclient::ApiClient;
+use cloudflare::framework::response::ApiFailure;
 
 use prettytable::{Cell, Row, Table};
 
 pub fn whoami(user: &GlobalUser) -> Result<(), failure::Error> {
-    // If using email + API key for auth, simply prints out email from config file.
+    // Attempt to print email for both GlobalKeyAuth and TokenAuth users
     let auth: String = match user {
         GlobalUser::GlobalKeyAuth { email, .. } => {
             format!("a Global API Key, associated with the email '{}'", email,)
         }
-        GlobalUser::TokenAuth { .. } => "an API Token".to_string(),
+        GlobalUser::TokenAuth { .. } => {
+            let token_auth_email: String = fetch_api_token_email(user)?;
+
+            if token_auth_email == "" {
+                "an API Token".to_string()
+            } else {
+                format!(
+                    "an API Token, associated with the email '{}'",
+                    token_auth_email,
+                )
+            }
```

Comment:
```suggestion
            if let Some(token_auth_email) = token_auth_email {
                format!(
                    "an API Token, associated with the email '{}'",
                    token_auth_email,
                )
            } else {
                "an API Token".to_string()
            }
```

Reply:
it's more idiomatic to use `Option::None` instead of assigning an empty string for an empty result